src/Resources/Loaders.h

```diff
@@ -0,0 +1,102 @@
+/*****************************************************************************
+ * Copyright (c) 2018-2022 openblack developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/openblack/openblack
+ *
+ * openblack is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#pragma once
+
+#include "3D/L3DAnim.h"
+#include "3D/L3DMesh.h"
+
+#include <PackFile.h>
+#include <entt/entt.hpp>
+
+namespace openblack::resources
+{
+
+using namespace l3d;
+
+template <typename Loader, typename Resource>
+struct BaseLoader: entt::resource_loader<Loader, Resource>
+{
+public:
+	using ResourceType = Resource;
+};
+
+struct L3DLoader final: BaseLoader<L3DLoader, L3DMesh>
+{
+	[[nodiscard]] std::shared_ptr<L3DMesh> load(const std::string& debugName, const std::vector<uint8_t>& data) const
+	{
+		auto mesh = std::make_shared<L3DMesh>(debugName);
+		if (!mesh->LoadFromBuffer(data))
+		{
+			throw std::runtime_error("Unable to load mesh");
+		}
+
+		return mesh;
+	}
+
+	[[nodiscard]] std::shared_ptr<L3DMesh> load(const std::string& debugName, const std::string& path) const
+	{
+		auto mesh = std::make_shared<L3DMesh>(debugName);
+		if (!mesh->LoadFromFile(path))
+		{
+			throw std::runtime_error("Unable to load mesh");
+		}
+
+		return mesh;
+	}
+};
+
+struct Texture2DLoader final: BaseLoader<Texture2DLoader, graphics::Texture2D>
```

Comment:
Could you add a loader for `.raw` textures from `filesystem.TexturePath()`

Reply:
Yeah I can do that. Should I do this in a new PR? This is getting big.

Reply:
Yes, could you make an issue for that?

Reply:
Added #369